### PR TITLE
Tweak unreachable lint wording

### PR DESCRIPTION
**Pull request by Nadrieril**
Some tweaks to the notes added in https://github.com/rust-lang/rust/pull/128034.

r? @estebank

### CODE REVIEW

File: tests/ui/pattern/usefulness/empty-types.exhaustive_patterns.stderr [hunk 1]
```diff
@@ -139,151 +147,169 @@ error: unreachable pattern
...
+   = note: to learn more about uninhabited types, see https://doc.rust-lang.org/nomicon/exotic-sizes.html#empty-types
 
 error: unreachable pattern
   --> $DIR/empty-types.rs:143:13
    |
 LL |             Some(_) => {}
    |             ^^^^^^^
    |
-   = note: this pattern matches no values because `Void` is uninhabited
+   = note: matches no values because `Void` is uninhabited
```
Comment (estebank) [inline tests/ui/pattern/usefulness/empty-types.exhaustive_patterns.stderr:222]: Should this maybe be a span label instead?

**Review by estebank — APPROVED** [review]
r=me with or without the requested change

File: tests/ui/pattern/usefulness/empty-types.exhaustive_patterns.stderr (hunk 1 above)
Comment (Nadrieril) [reply tests/ui/pattern/usefulness/empty-types.exhaustive_patterns.stderr]: Good idea! It's done.

**Pull request merged**